tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/field/FieldElementUtils.java

```diff
@@ -0,0 +1,192 @@
+package dk.alexandra.fresco.tools.mascot.field;
+
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+import dk.alexandra.fresco.tools.mascot.arithm.ArithmeticCollectionUtils;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class FieldElementUtils extends ArithmeticCollectionUtils<FieldElement> {
+
+  private final BigInteger modulus;
+  private final int modBitLength;
+  private final List<FieldElement> generators;
+
+  /**
+   * Creates new {@link FieldElementUtils}.
+   * 
+   * @param modulus modulus for underlying field element operations
+   * @param modBitLength modulus bit length
+   */
+  public FieldElementUtils(BigInteger modulus, int modBitLength) {
+    super();
+    this.modulus = modulus;
+    this.modBitLength = modBitLength;
```

Comment:
I think modBitLength should be computed from the modulus it self.

Reply:
I could have sworn `BigInteger.bitLength` does not give us that information but apparently it does (at least for a correctly chosen modulus that does actually require modBitLength bits to). That will make a lot of the code substantially cleaner.

Reply:
Done.